cmd/commands/common.go

```diff
@@ -200,10 +200,18 @@ func getRuntimeNameFromUserSelect(ctx context.Context) (string, error) {
 		return "", fmt.Errorf("no runtimes were found")
 	}
 
-	runtimeNames := make([]string, len(runtimes))
+	var runtimeNames []string
 
-	for index, rt := range runtimes {
-		runtimeNames[index] = rt.Metadata.Name
+	for _, rt := range runtimes {
+		rtDisplay := rt.Metadata.Name
+		if rt.Managed {
+			if cmdFlow == RuntimeUpgrade {
```

Comment:
I think instead of this whole cmdFlow type, which seems a bit overkill, you can just pass along `allowManaged bool`. You can even do it as optional args with the spread operator so it will not break existing calls, and you can just pass `true` from flows where you want to list managed runtimes.

Reply:
Another good reason why not to do it like this is that the decision of which flow is allowed for managed and which flow isn't should definitely not be in `getRuntimeNameFromUserSelect`.